plugin-markerview/src/main/java/com/mapbox/mapboxsdk/plugins/markerview/MarkerViewManager.java

```diff
@@ -0,0 +1,67 @@
+package com.mapbox.mapboxsdk.plugins.markerview;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.UiThread;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import timber.log.Timber;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MarkerViewManager implements MapView.OnMapChangedListener {
+
+  private final MapView mapView;
+  private final MapboxMap mapboxMap;
+  private final List<MarkerView> markers = new ArrayList<>();
+  private boolean initalised;
+
+  public MarkerViewManager(MapView mapView, MapboxMap mapboxMap) {
+    this.mapView = mapView;
+    this.mapboxMap = mapboxMap;
+  }
+
+  public void onDestroy() {
+    mapView.removeOnMapChangedListener(this);
+    initalised = false;
+  }
+
+  @UiThread
+  public void addMarker(@NonNull MarkerView markerView) {
+    if (markers.contains(markerView)) {
```

Comment:
We should check here for `MapView#isDestroyed`, not to hook the listener into a destroyed map.